operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java

```diff
@@ -89,12 +87,6 @@ public final void registerEventSource(String name, EventSource eventSource)
         // leave untouched
         throw e;
       }
-      if (e instanceof KubernetesClientException) {
-        KubernetesClientException ke = (KubernetesClientException) e;
-        if (404 == ke.getCode()) {
-          throw new MissingCRDException(null, null);
-        }
-      }
```

Comment:
@csviri @metacosm removed this because a 404 may also be the result of watching a resource by name or in a namespace that does not exists so assuming 404 is only when a CRD is missing seems wrong